Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/ServiceFacadeInspector.cs

```diff
@@ -0,0 +1,405 @@
+                }
+
+                GUILayout.FlexibleSpace();
+                GUILayout.EndHorizontal();
+                EditorGUILayout.Space();
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Draws a list of services that use this as a data provider
+        /// </summary>
+        /// <param name="serviceType"></param>
+        /// <returns></returns>
+        private bool DrawDataProviders(Type serviceType)
+        {  
+            // If this is a data provider being used by other services, mention that now
+            dataProviderList.Clear();
+            foreach (MixedRealityDataProviderAttribute dataProviderAttribute in serviceType.GetCustomAttributes(typeof(MixedRealityDataProviderAttribute), true))
+            {
+                dataProviderList.Add(" • " + dataProviderAttribute.ServiceInterfaceType.Name);
+            }
+
+            if (dataProviderList.Count > 0)
+            {
+                EditorGUILayout.HelpBox("This data provider is used by the following services:\n " + String.Join("\n", dataProviderList.ToArray()), MessageType.Info);
+                EditorGUILayout.Space();
+                return true;
+            }
+            return false;
+        }
+               
+        /// <summary>
+        /// Draws the custom inspector gui for service type.
+        /// </summary>
+        /// <param name="facade"></param>
+        /// <returns></returns>
+        private bool DrawInspector(ServiceFacade facade)
+        {
+            IMixedRealityServiceInspector inspectorInstance;
+            if (GetServiceInspectorInstance(facade.ServiceType, out inspectorInstance))
+            {
+                inspectorInstance.DrawInspectorGUI(facade.Service);
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Draws the profile for the service type, if wanted by inspector and found.
+        /// </summary>
+        /// <param name="serviceType"></param>
+        /// <returns></returns>
+        private bool DrawProfile(Type serviceType)
+        {
+
+            IMixedRealityServiceInspector inspectorInstance;
+            if (GetServiceInspectorInstance(serviceType, out inspectorInstance))
+            {
+                if (!inspectorInstance.DrawProfileField)
+                {   // We've been instructed to skip drawing a profile by the inspector
+                    return false;
+                }
+            }
+
+            bool foundAndDrewProfile = false;
+
+            // Draw the base profile stuff
+            if (typeof(BaseCoreSystem).IsAssignableFrom(serviceType))
+            {
+                SerializedObject activeProfileObject = new SerializedObject(MixedRealityToolkit.Instance.ActiveProfile);
+                // Would be nice to handle this using some other method
+                // Would be nice to handle this with a lookup instead
```

Comment:
>// Would be nice to handle this with a lookup instead [](start = 16, length = 53)

We could create a const/final static dictionary for this class that has

Dictionary<Type, string> = {
  {IMixedRealityInputSystem, "inputSystemProfile"},
  {IMixedRealityBoundarySystem, "boundaryVisualizationProfile"},
 ....
}

And then use that to lookup the property name.

Since it looks like the only delta here is the property name.

Alternatively we could just move:

BaseMixedRealityProfileInspector.RenderProfile(serviceProfileProp, null, false, serviceType);
                    EditorGUILayout.Space();
                    foundAndDrewProfile = true;

Out of each if/elseif statement and into another block like this:

SerializedObject activeProfileObject = new SerializedObject(MixedRealityToolkit.Instance.ActiveProfile);
                // Would be nice to handle this using some other method
                // Would be nice to handle this with a lookup instead
SerializedProperty serviceProfileProp;
                if (typeof(IMixedRealityInputSystem).IsAssignableFrom(serviceType))
                {
                    serviceProfileProp = activeProfileObject.FindProperty("inputSystemProfile");
                    foundAndDrewProfile = true;
                }
                else if (typeof(IMixedRealityBoundarySystem).IsAssignableFrom(serviceType))
                {
                    serviceProfileProp = activeProfileObject.FindProperty("boundaryVisualizationProfile");
                    foundAndDrewProfile = true;
                }
                else if (typeof(IMixedRealityDiagnosticsSystem).IsAssignableFrom(serviceType))
                {
                    serviceProfileProp = activeProfileObject.FindProperty("diagnosticsSystemProfile");
                    foundAndDrewProfile = true;
                }
                else if (typeof(IMixedRealitySpatialAwarenessSystem).IsAssignableFrom(serviceType))
                {
                    serviceProfileProp = activeProfileObject.FindProperty("spatialAwarenessSystemProfile");
                    foundAndDrewProfile = true;
                }

if (foundAndDrewProfile)
{
  BaseMixedRealityProfileInspector.RenderProfile(serviceProfileProp, null, false, serviceType);
                    EditorGUILayout.Space();
                    foundAndDrewProfile = true;
}

Reply:
I'm marking this as resolved but I'd like to revisit it in the future.